code/game/machinery/firealarm.dm

```diff
@@ -38,23 +38,52 @@ FIRE ALARM
 
 	var/last_time_pulled //used to prevent pulling spam by same persons
 
-/obj/machinery/firealarm/no_alarm
-	report_fire_alarms = FALSE
+/obj/machinery/firealarm/New(location, direction, building)
+	. = ..()
 
-/obj/machinery/firealarm/syndicate
-	report_fire_alarms = FALSE
-	show_alert_level = FALSE
+	if(building)
+		buildstage = 0
+		wiresexposed = TRUE
+		setDir(direction)
+		set_pixel_offsets_from_dir(26, -26, 26, -26)
+
+	if(is_station_contact(z) && show_alert_level)
+		if(GLOB.security_level)
+			overlays += image('icons/obj/monitors.dmi', "overlay_[get_security_level()]")
+		else
+			overlays += image('icons/obj/monitors.dmi', "overlay_green")
```

Comment:
`add_overlay` over `overlays +=`. Also don't need to do `image()` or specify the icon file since it's the same `icon` of the firelock.
```suggestion
		if(GLOB.security_level)
			add_overlay("overlay_[get_security_level()]")
		else
			add_overlay("overlay_green")
```